src/mem/cache/replacement_policies/tree_plru_rp.cc

```diff
@@ -104,8 +104,6 @@ TreePLRU::TreePLRUReplData::TreePLRUReplData(
 TreePLRU::TreePLRU(const Params &p)
   : Base(p), numLeaves(p.num_leaves), count(0), treeInstance(nullptr)
 {
-    fatal_if(!isPowerOf2(numLeaves),
```

Comment:
It must be greater than 0 because of the modulus operator

Reply:
There is also a bug when num_leaves == 1: lines 203 and 208 initialize index as 0, and line 129 does 0-1/2

Reply:
I added a check that the number of leaves is >= 2 for now. I confirmed that it crashes if numLeaves == 1.